feat(plugin-history-sync): sort routes by variable count

tonyfromundefined opened the pull request.

tonyfromundefined commented on extensions/plugin-history-sync/src/makeTemplate.ts:
```diff
@@ -95,5 +95,6 @@ export function makeTemplate(
         ...pathParams,
       };
     },
+    variableCount: (pattern as any).names.length,
```
Use unstable API in `url-pattern` 😅

orionmiz commented on extensions/plugin-history-sync/src/sortRoutes.ts:
```diff
@@ -0,0 +1,33 @@
+  const routes: Route[] = [];
+
+  for (const activityName of activityNames) {
+    const templateStrs = normalizeRoute(routeMap[activityName]);
+
+    for (const templateStr of templateStrs) {
+      routes.push({
+        activityName,
+        templateStr,
+      });
+    }
```
`sortRoutes()` is having `normalizedRoute()` and generates `Route` inside.. it seems a little bit ambigious for me.

It would be more clearer if a new function called `normalizeRouteMap()`
 performs making `Route[]` and return them.

```ts
export function normalizeRouteMap<
  T extends {
    [activityName: string]: string | string[];
  },
>(routeMap: T) {
  const activityNames = Object.keys(routeMap);
  const routes: Route[] = [];

  for (const activityName of activityNames) {
    const templateStrs = normalizeRoute(routeMap[activityName]);

    for (const templateStr of templateStrs) {
      routes.push({
        activityName,
        templateStr,
      });
    }
  }

  return routes;
}
```

Then, we can just call `sortRoutes()` function like:
```ts
const routes = normalizeRouteMap(options.routes)
const sortedRoutes = sortRoutes(routes);
```

tonyfromundefined commented:
I refactored everything using the `normalizeRouteMap()` you suggested. I removed all the logic previously used using `options.routes` and changed it to process using a normalized route array.

orionmiz reviewed (approved):
👍
